contracts/axelar-interchain-token-service/src/test.rs

```diff
@@ -0,0 +1,149 @@
+    let trusted_address = String::from_str(&env, "trusted_address");
+
+    client.mock_auths(&[MockAuth {
+        address: &not_owner,
+        invoke: &MockAuthInvoke {
+            contract: &contract_id,
+            fn_name: "set_trusted_address",
+            args: (chain.clone(), trusted_address.clone()).into_val(&env),
+            sub_invokes: &[],
+        },
+    }]).set_trusted_address(&chain, &trusted_address);
+
+}
+
+#[test]
+fn set_trusted_address() {
+    let (env, contract_id, client) = setup_env();
+    let owner = Address::generate(&env);
+
+    initialize_its(&env, &client, owner.clone());
+
+    let chain = String::from_str(&env, "chain");
+    let trusted_address = String::from_str(&env, "trusted_address");
+    
+    client.mock_auths(&[MockAuth {
+        address: &owner,
+        invoke: &MockAuthInvoke {
+            contract: &contract_id,
+            fn_name: "set_trusted_address",
+            args: (chain.clone(), trusted_address.clone()).into_val(&env),
+            sub_invokes: &[],
+        },
+    }]).set_trusted_address(&chain, &trusted_address);
+
+    assert_eq!(
+        client.is_trusted_address(&chain, &trusted_address),
+        true
+    );
+}
+
+#[test]
+fn remove_trusted_address() {
+    let (env, contract_id, client) = setup_env();
+    let owner = Address::generate(&env);
+
+    initialize_its(&env, &client, owner.clone());
+
+    let chain = String::from_str(&env, "chain");
+    let trusted_address = String::from_str(&env, "trusted_address");
+    
+    client.mock_auths(&[MockAuth {
+        address: &owner,
+        invoke: &MockAuthInvoke {
+            contract: &contract_id,
+            fn_name: "set_trusted_address",
+            args: (chain.clone(), trusted_address.clone()).into_val(&env),
+            sub_invokes: &[],
+        },
+    }]).set_trusted_address(&chain, &trusted_address);
+
+    client.mock_auths(&[MockAuth {
+        address: &owner,
+        invoke: &MockAuthInvoke {
+            contract: &contract_id,
+            fn_name: "remove_trusted_address",
+            args: (chain.clone(), trusted_address.clone()).into_val(&env),
+            sub_invokes: &[],
+        },
+    }]).remove_trusted_address(&chain, &trusted_address);
+
+    assert_eq!(
+        client.is_trusted_address(&chain, &trusted_address),
+        false
+    );
+}
+
+#[test] 
+fn transfer_ownership() {
+    let (env, contract_id, client) = setup_env();
+    let owner = Address::generate(&env);
+
+    initialize_its(&env, &client, owner.clone());
+
+    let new_owner = Address::generate(&env);
+
+    client.mock_auths(&[MockAuth {
+        address: &owner,
+        invoke: &MockAuthInvoke {
+            contract: &contract_id,
+            fn_name: "transfer_ownership",
+            args: (&new_owner,).into_val(&env),
+            sub_invokes: &[],
+        },
+    }]).transfer_ownership(&new_owner);
+
+    assert_eq!(
+        client.owner(),
+        new_owner
```

Comment:
you can check the codecov links to ensure coverage for the PR is always 100%

Reply:
Codecov complaining about attributes required by Soroban

Reply:
yeah, those are expected, our tests might not make use of something like Debug trait of that type etc. which doesn't require coverage. We care more about the coverage of the logic